listener/src/event_handler/flux_transition_handler.rs

```diff
@@ -0,0 +1,209 @@
+        let existing_state = State::new(H256::zero(), U256::zero(), vec![0, 0, 0, 0], 1);
+        let entity = existing_state.into();
+        store.apply_entity_operations(vec![EntityOperation::Set {
+            key: util::entity_key("AccountState", &entity),
+            data: entity
+        }], None).unwrap();
+
+        let first_deposit = PendingFlux {
+            slot_index: 0,
+            slot: U256::zero(),
+            account_id: 0,
+            token_id: 0,
+            amount: 10,
+        };
+        let mut entity: Entity = first_deposit.into();
+        entity.set("id", "1");
+        store.apply_entity_operations(vec![EntityOperation::Set {
+            key: util::entity_key("Deposit", &entity),
+            data: entity
+        }], None).unwrap();
+
+        let second_deposit = PendingFlux {
+            slot_index: 1,
+            slot: U256::zero(),
+            account_id: 1,
+            token_id: 0,
+            amount: 10,
+        };
+        let mut entity: Entity = second_deposit.into();
+        entity.set("id", "2");
+        store.apply_entity_operations(vec![EntityOperation::Set {
+            key: util::entity_key("Deposit", &entity),
+            data: entity
+        }], None).unwrap();
+
+        // Process event
+        let log = create_state_transition_event(FluxTransitionType::Deposit, 1, H256::from(1), 0);
+        let result = handler.process_event(
+            util::test::logger(), 
+            Arc::new(util::test::fake_block()),
+            Arc::new(util::test::fake_tx()), 
+            log
+        );
+        let expected_new_state = State::new(H256::from(1), U256::one(), vec![10, 10, 0, 0], 1);
+
+        assert!(result.is_ok());
+        match result.unwrap().pop().unwrap() {
+            EntityOperation::Set { key: _, data } => assert_eq!(State::from(data), expected_new_state),
+            _ => assert!(false)
+        }
+    }
+
+    #[test]
+    fn test_fails_if_state_does_not_exist() {
+        let schema = util::test::fake_schema();
+        let store = Arc::new(MockStore::new(vec![(schema.id.clone(), schema)]));
+        let handler = FluxTransitionHandler::new(store.clone());
+
+        // No data in store
+
+        let log = create_state_transition_event(FluxTransitionType::Deposit, 1, H256::from(1), 0);
+        let result = handler.process_event(
+            util::test::logger(), 
+            Arc::new(util::test::fake_block()),
+            Arc::new(util::test::fake_tx()), 
+            log
+        );
+        assert!(result.is_err());
+    }
+
+    fn create_state_transition_event(
+        transition_type: FluxTransitionType, 
+        new_state_index: u8, 
+        new_state_root: H256,
+        applied_slot: u8
+    ) -> Arc<Log> {
+        let bytes: Vec<Vec<u8>> = vec![
+            /* transition_type */ vec![0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, transition_type as u8],
```

Comment:
Is there no cleaner way to construct these arrays?

Reply:
Happy to suggestions. If I assign each vector to it's own variable it become really annoying merging them (it requires one extra line for each variable). I found this flatMap style the most clean while still expressive, but agree it could be better.